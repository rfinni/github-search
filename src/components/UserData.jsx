import React from 'react';
import PropTypes from 'prop-types';

const UserData = ({ user }) => (
  <div className="user-data">
    <span className="name">{user.name} ({user.login})</span>
    <span>{user.location}</span>
    <span>Joined GitHub in {new Date(user.joined).getFullYear()}</span>
  </div>
);

UserData.propTypes = {
  user: PropTypes.object.isRequired,
};

export default UserData;
