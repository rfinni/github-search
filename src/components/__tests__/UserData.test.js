import React from 'react';
import renderer from 'react-test-renderer';
import UserData from '../UserData';
import { stubUser } from '../../../testing/stubs';

describe('UserData.jsx', () => {
  const wrapper = renderer.create(<UserData user={stubUser} />).toJSON();
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

