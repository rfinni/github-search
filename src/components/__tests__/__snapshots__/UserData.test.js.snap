// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserData.jsx should render correctly 1`] = `
<div
  className="user-data"
>
  <span
    className="name"
  >
    Ryan Finni
     (
    rfinni
    )
  </span>
  <span>
    Chicago, IL
  </span>
  <span>
    Joined GitHub in 
    NaN
  </span>
</div>
`;
